### PR TITLE
Add more dartdoc to widgets.dart

### PR DESCRIPTION
**Pull request by abarth**
This patch includes documentation for transitions.dart and for
scrollable.dart.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/widgets/scrollable.dart [hunk 1]
```diff
@@ -50,13 +56,25 @@ abstract class Scrollable extends StatefulComponent {
...
   final ScrollListener onScroll;
+
+  /// Called whenever this widget stops scrolling.
   final ScrollListener onScrollEnd;
+
+  /// Called to determine the offset to which scrolling should snap.
   final SnapOffsetCallback snapOffsetCallback;
-  final double snapAlignmentOffset;
+
+  final double snapAlignmentOffset; // What does this do?
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scrollable.dart]: find out :-)

Comment (Hixie) [inline packages/flutter/lib/src/widgets/scrollable.dart]: @hansmuller probably knows

File: packages/flutter/lib/src/widgets/transitions.dart [hunk 2]
```diff
@@ -14,6 +14,13 @@ import 'framework.dart';
 export 'package:flutter/animation.dart' show AnimationDirection;
 export 'package:flutter/rendering.dart' show RelativeRect;
 
+/// A component that rebuilds when the given animation changes value.
+///
+/// AnimatedComponent, is most useful for stateless animated widgets. To use
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/transitions.dart]: extraneous comma

File: packages/flutter/lib/src/widgets/transitions.dart [hunk 3]
```diff
@@ -145,6 +195,7 @@ class FadeTransition extends AnimatedComponent {
   }
 }
 
+/// Animates the background color of a widget.
 class ColorTransition extends AnimatedComponent {
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/transitions.dart]: I am conflicted about this class.

File: packages/flutter/lib/src/widgets/transitions.dart [hunk 4]
```diff
@@ -203,8 +261,28 @@ class PositionedTransition extends AnimatedComponent {
   }
 }
 
+/// A builder that builds a widget given a child.
 typedef Widget TransitionBuilder(BuildContext context, Widget child);
 
+/// A general-purpose widget for building animations.
+///
+/// AnimatedBuilder, is useful for more complex components that wish to include
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/transitions.dart]: extra comma

**Comment by Hixie** [comment]
lgtm

File: packages/flutter/lib/src/widgets/scrollable.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/lib/src/widgets/scrollable.dart]: I looked into it and didn't quite understand what it was doing.  Hopefully @HansMuller can explain it to me and we can document it.

File: packages/flutter/lib/src/widgets/transitions.dart (hunk 3 above)
Comment (abarth) [inline packages/flutter/lib/src/widgets/transitions.dart]: Looks like it was a primitive version of ModalBarrier from the fn2 days.  I'll post a separate patch to remove it.

**Pull request merged**